samples/bluetooth/channel_sounding/src/distance_estimation.c

```diff
@@ -0,0 +1,303 @@
+				theta[j] += 2.0f * PI;
+			}
+		}
+	}
+
+	float phase_slope = linear_regression(frequencies, theta, num_angles);
+
+	float distance = -phase_slope * (SPEED_OF_LIGHT_M_PER_S / (4 * PI));
+
+	return distance / 1000000.0f; /* Scale to meters. */
+}
+
+static float estimate_distance_using_time_of_flight(uint8_t n_samples)
+{
+	float tof;
+	float tof_mean = 0.0;
+
+	/* Cumulative Moving Average */
+	for (uint8_t i = 0; i < n_samples; i++) {
+		if (!mode_1_data[i].failed) {
+			tof = (mode_1_data[i].toa_tod_initiator -
+			       mode_1_data[i].tod_toa_reflector) /
+			      2;
+			tof_mean += (tof - tof_mean) / (i + 1);
+		}
+	}
+
+	float tof_mean_ns = tof_mean / 2.0f;
+
+	return tof_mean_ns * SPEED_OF_LIGHT_NM_PER_S;
+}
+
+static bool process_step_data(struct bt_le_cs_subevent_step *step, void *user_data)
+{
+	struct processing_context *context = (struct processing_context *)user_data;
+
+	if (step->mode == BT_CONN_LE_CS_MAIN_MODE_2) {
+		struct bt_hci_le_cs_step_data_mode_2 *step_data =
+			(struct bt_hci_le_cs_step_data_mode_2 *)step->data;
+
+		if (context->local_steps) {
+			for (uint8_t i = 0; i < (context->n_ap + 1); i++) {
+				if (step_data->tone_info[i].extension_indicator !=
+				    BT_HCI_LE_CS_NOT_TONE_EXT_SLOT) {
+					continue;
+				}
+
+				mode_2_data[context->mode_2_data_index].channel = step->channel;
+				mode_2_data[context->mode_2_data_index].antenna_permutation =
+					step_data->antenna_permutation_index;
+				mode_2_data[context->mode_2_data_index].local_iq_sample =
+					bt_le_cs_parse_pct(
+						step_data->tone_info[i].phase_correction_term);
+				if (step_data->tone_info[i].quality_indicator ==
+					    BT_HCI_LE_CS_TONE_QUALITY_LOW ||
+				    step_data->tone_info[i].quality_indicator ==
+					    BT_HCI_LE_CS_TONE_QUALITY_UNAVAILABLE) {
+					mode_2_data[context->mode_2_data_index].failed = true;
+				}
+
+				context->mode_2_data_index++;
+			}
+		} else {
+			for (uint8_t i = 0; i < (context->n_ap + 1); i++) {
+				if (step_data->tone_info[i].extension_indicator !=
+				    BT_HCI_LE_CS_NOT_TONE_EXT_SLOT) {
+					continue;
+				}
+
+				mode_2_data[context->mode_2_data_index].peer_iq_sample =
+					bt_le_cs_parse_pct(
+						step_data->tone_info[i].phase_correction_term);
+				if (step_data->tone_info[i].quality_indicator ==
+					    BT_HCI_LE_CS_TONE_QUALITY_LOW ||
+				    step_data->tone_info[i].quality_indicator ==
+					    BT_HCI_LE_CS_TONE_QUALITY_UNAVAILABLE) {
+					mode_2_data[context->mode_2_data_index].failed = true;
+				}
+
+				context->mode_2_data_index++;
+			}
+		}
+	} else if (step->mode == BT_HCI_OP_LE_CS_MAIN_MODE_1) {
+		struct bt_hci_le_cs_step_data_mode_1 *step_data =
+			(struct bt_hci_le_cs_step_data_mode_1 *)step->data;
+
+		if (step_data->packet_quality_aa_check !=
+			    BT_HCI_LE_CS_PACKET_QUALITY_AA_CHECK_SUCCESSFUL ||
+		    step_data->packet_rssi == BT_HCI_LE_CS_PACKET_RSSI_NOT_AVAILABLE ||
+		    step_data->tod_toa_reflector == BT_HCI_LE_CS_TIME_DIFFERENCE_NOT_AVAILABLE) {
```

Comment:
`BT_HCI_LE_CS_TIME_DIFFERENCE_NOT_AVAILABLE` is out of the range of `int16_t tod_toa_reflector`

Reply:
I'll change it back to uint16_t in hci_types.h